src/main/java/games/strategy/util/Version.java

```diff
@@ -131,58 +138,34 @@ public int hashCode() {
   }
 
   @Override
-  public int compareTo(final Version o) {
-    return o == null ? -1 : compareTo(o, false);
-  }
+  public int compareTo(final Version other) {
+    checkNotNull(other);
 
-  private int compareTo(final Version other, final boolean ignoreMicro) {
-    if (other == null) {
-      return -1;
-    }
-    if (other.m_major > m_major) {
-      return 1;
-    }
-    if (other.m_major < m_major) {
-      return -1;
-    } else if (other.m_minor > m_minor) {
-      return 1;
-    } else if (other.m_minor < m_minor) {
-      return -1;
-    } else if (other.m_point > m_point) {
-      return 1;
-    } else if (other.m_point < m_point) {
-      return -1;
-    } else if (!ignoreMicro) {
-      if (other.m_micro > m_micro) {
-        return 1;
-      } else if (other.m_micro < m_micro) {
-        return -1;
-      }
-    }
-    // if the only difference is m_micro, then ignore
-    return 0;
+    return compareTo(other, NO_COMPARE_OPTIONS);
```

Comment:
I wondered why you're explicitly passing an empty array to this method, given that the argument is optional.

Reply:
(I figured out why, still seems weird to me)